CHANGELOG.md

```diff
@@ -32,6 +32,22 @@
     end
     ```
 
+- Support `Sentry::Transaction#set_measurement` [#1838](https://github.com/getsentry/sentry-ruby/pull/1838)
+
+    Usage:
+
+    ```rb
+    # enable this experimental feature first
+    Sentry.init do |config|
+      # other configs
+      config.experiments.custom_measurements = true
+    end
+
+    # in your app
+    transaction = Sentry.get_current_scope.get_transaction
```

Comment:
we can introduce a top level `Sentry.set_measurement` too that fetches the transaction automatically from the scope.

Not sure what you think about the `Sentry.measure` API you suggested but those enhancements can probably be done separately. I know someone that wanted to make a memory measuring proof-of-concept, so we could do it along with that?

Reply:
I think that makes sense but prefer leaving it to another PR, as this one focuses on the foundation API and configuration. WDYT?

Reply:
yep, let's do it separately.